pkg/report/table/summary.go

```diff
@@ -0,0 +1,303 @@
+
+func (s MisconfigScanner) Header() string {
+	return "Misconfigurations"
+}
+
+func (s MisconfigScanner) Count(result types.Result) int {
+	if result.Class == types.ClassConfig {
+		return len(result.Misconfigurations)
+	}
+	return -1
+}
+
+func (s MisconfigScanner) String() string {
+	return string(types.MisconfigScanner)
+}
+
+type SecretScanner struct{ scannerAlignment }
+
+func (s SecretScanner) Header() string {
+	return "Secrets"
+}
+
+func (s SecretScanner) Count(result types.Result) int {
+	if result.Class == types.ClassSecret {
+		return len(result.Secrets)
+	}
+	return -1
+}
+
+func (s SecretScanner) String() string {
+	return string(types.SecretScanner)
+}
+
+type LicenseScanner struct{ scannerAlignment }
+
+func (s LicenseScanner) Header() string {
+	return "Licenses"
+}
+
+func (s LicenseScanner) Count(result types.Result) int {
+	if result.Class == types.ClassLicense || result.Class == types.ClassLicenseFile {
+		return len(result.Licenses)
+	}
+	return -1
+}
+
+func (s LicenseScanner) String() string {
+	return string(types.LicenseScanner)
+}
+
+type summaryRenderer struct {
+	w          *bytes.Buffer
+	isTerminal bool
+	scanners   []Scanner
+	logger     *log.Logger
+}
+
+func NewSummaryRenderer(buf *bytes.Buffer, isTerminal bool, scanners types.Scanners) *summaryRenderer {
+	if !isTerminal {
+		tml.DisableFormatting()
+	}
+
+	var ss []Scanner
+	for _, scanner := range scanners {
+		s := NewScanner(scanner)
+		if lo.IsNil(s) {
+			continue
+		}
+		ss = append(ss, s)
+	}
+
+	return &summaryRenderer{
+		w:          buf,
+		isTerminal: isTerminal,
+		scanners:   ss,
+		logger:     log.WithPrefix("report"),
+	}
+}
+
+func (r *summaryRenderer) Render(report types.Report) {
+	if len(r.scanners) == 0 {
+		r.logger.Warn("No enabled scanners found. Summary table will not be displayed.")
+		return
+	}
+
+	r.printf("\n<underline><bold>Report Summary</bold></underline>\n\n")
+
+	t := newTableWriter(r.w, r.isTerminal)
+	t.SetAutoMerge(false)
+	t.SetColumnMaxWidth(80)
+
+	headers := []string{
+		"Target",
+		"Type",
+	}
+	alignments := []table.Alignment{
+		table.AlignLeft,
+		table.AlignCenter,
+	}
+	for _, scanner := range r.scanners {
+		headers = append(headers, scanner.Header())
+		alignments = append(alignments, scanner.Alignment())
+	}
+	t.SetHeaders(headers...)
+	t.SetAlignment(alignments...)
+
+	for _, result := range splitAggregatedPackages(report.Results) {
+		resultType := string(result.Type)
+		if result.Class == types.ClassSecret {
+			resultType = "text"
+		} else if result.Class == types.ClassLicense || result.Class == types.ClassLicenseFile {
+			resultType = "-"
+		}
+		rows := []string{
+			result.Target,
+			resultType,
+		}
+		for _, scanner := range r.scanners {
+			rows = append(rows, r.colorizeCount(scanner.Count(result)))
+		}
+		t.AddRows(rows)
+	}
+
+	if len(report.Results) == 0 {
+		r.showEmptyResultsWarning()
+		t.AddRows(slices.Repeat([]string{"-"}, len(r.scanners)+2))
+	}
```

Comment:
nit: It's no big deal, but the alignment looks a bit strange when `-` is used in all columns.

```
┌────────┬──────┬─────────────────┬─────────┐
│ Target │ Type │ Vulnerabilities │ Secrets │
├────────┼──────┼─────────────────┼─────────┤
│ -      │  -   │        -        │    -    │
└────────┴──────┴─────────────────┴─────────┘
```

```suggestion
	if len(report.Results) == 0 {
		r.showEmptyResultsWarning()
		alignments[0] = table.AlignCenter
		t.SetAlignment(alignments...)
		t.AddRows(slices.Repeat([]string{"-"}, len(r.scanners)+2))
	}
```